### DISCUSSION TITLE
SEO & Site Search is missing a setting for Site Alias Mapping Mode

### DISCUSSION THREAD

**Discussion by eswanzey** [vanjarosoftware/Vanjaro.Platform#2119] [General]
I believe that Site Alias Mapping Mode (Canonical, Redirect, None) is missing from the Vanjaro settings panel. A screenshot is attached of what I'm talking about
![2022-03-02_07-33-40](https://user-images.githubusercontent.com/8103212/156363505-6530f33a-bf88-4b45-b1b7-fdef996f546b.jpg)
.

**Comment by meetmandeep** [comment]
Adding to the roadmap to appear in "Domains"

**Comment by eswanzey** [comment]
I took a DNN site, hybrid site, and Vanjaro Business site and changed PortalSettings.PortalAliasMapping from REDIRECT to CANONICALURL. From what I'm seeing, it didn't change a thing. I was expecting to see a link element in the head section with rel="canonical" and defined href. I also used Google's Seaech Console to see what it looked like to a spider, and same result, there still was no link defined.

I don't see any mention in DNN or Vanjaro about this. I believe that early DNN 9 started setting it to CANONICALURL by default. At this point I would have to call it a bug on both platforms unless someone can shed some light!

**Comment by meetmandeep** [comment]
Vanjaro uses Redirect by default and the UI was purposely omitted. Can you share your use case for Canonical?

**Comment by eswanzey** [comment]
I can see why Redirect is the default in Vanjaro because Vanjaro starts from a clean slate.

Migrated sites (like DNN, but true for any) sometimes have 4+ redirects per page on very old pages, and old pages can of course have significant SEO value. So, the use case is that when there are excessive redirects (say, more than 2 or 3), being able to specify canonical becomes important because there are so many 301 redirects going on to get to the ultimate URL. They also don’t always happen within the platform, e.g., I have many sites where there might be 301s that exist for a page within the DNN context, then more that get defined within a web.config (i.e., outside the context of the CMS).

When I think about this, I don’t see why pages just don’t have the canonical link in the header of every page anyway. It’s programmatically insignificant, and can only do good, not harm.

**Comment by meetmandeep** [comment]
I may be missing something; but the canonical option in DNN is at Portal Alias level and does not consider individual 301 redirects of pages. It's there so you can run same website of two different URLs www.siteA.com and www.siteB.com with one of time pointing a canonical reference to the primary alias. When developing Vanjaro, we did not see a use case for this anymore on new sites. If you're migrating a site from DNN or running Vanjaro for DNN then you do have the DNN UI available. 

After discussion with our team internally, we do not foresee introducing this option. There can only be a single primary domain for a website; again, the setting in question is a Portal Alias/Domain setting and not specific to individual pages with multiple 301 redirects.

**Comment by Tychodewaard** [comment]
If you have different urls/domains that you want to use but want to prevent duplicate content, you want the canonical option.

**Reply by meetmandeep** [reply]
@Tychodewaard I totally understand the canonical option. The question is why would you want different urls/domains for the same site? The only use case I can think of is a rare possibility of changing domain names on an already established site that has a fairly large SERP/SEO value; even then, why canonical and not just 301 Redirect?
